### ISSUE TITLE
Assertion error: "The valueProperty of the focusedSoccerBall should not be null."

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/center-and-variability#574]
**Test device**
MacBook Air M1 chip

**Operating System**
14.0

**Browser**
Safari 17

**Problem description**
While verifying an issue on main, the sim froze when I pressed the Reset All button with the mouse while a soccer ball had focus. I don't see this issue in https://github.com/phetsims/qa/issues/985. 

**Steps to reproduce**
1. On the Median Screen, Press the Kick 5 button
2. Tab to and grab a soccer ball
3. While the ball has focus, press Reset All with the mouse

**Visuals**
<img width="512" alt="Screenshot 2023-10-03 at 9 55 36 AM" src="https://github.com/phetsims/center-and-variability/assets/87318828/cf1210e7-3d90-43f0-a74e-186edca360a5">

<img width="512" alt="Screenshot 2023-10-03 at 9 55 54 AM" src="https://github.com/phetsims/center-and-variability/assets/87318828/e30a87bc-ddb8-45cf-8a6d-e30ab8c7b14d">

**Comment by marlitas** [comment]
This was caused by a commit fixing a different bug in: https://github.com/phetsims/center-and-variability/issues/571.

This situation should now be addressed. We have to reset the focusedSoccerBallProperty before resetting the sceneModels. I made sure to move other keyboard focus related Properties before the sceneModel reset as well since it makes sense that we want to reset our general Properties before triggering reset logic in the sceneModels that rely on those Properties.

Over to @samreid and @matthew-blackman for review.

@Nancy-Salpepi this should also be ready to check on main.
